src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -1029,7 +1029,16 @@ private static unsafe void UpdatedCachedCursorPosition(byte* bufPtr, int count,
                     byte c = bufPtr[i];
                     if (c < 127 && c >= 32) // ASCII/UTF-8 characters that take up a single position
                     {
-                        IncrementX();
+                        left++;
+
+                        // After printing in the last column, setting CursorLeft is expected to
+                        // place the cursor back in that same row.
+                        // Invalidate the cursor position rather than moving it to the next row.
```

Comment:
Is this because behavior varies and so we can't predict what the terminal is actually going to do?

Reply:
We need to treat one position past the window width special in that setting `CursorLeft` from that will remain on the same line.

The position behaves special.
```cs
Console.CursorLeft = Console.BufferWidth - 3;
var positions = new (int Left, int Top)[5];
for (int i = 0; i < positions.Length; i++)
{
    Console.Write((char)0xA3);
    positions[i] = Console.GetCursorPosition();
}
Console.WriteLine();
Console.WriteLine(string.Join(" ", positions));
```
When I run this, I get:
```
                                                                                                                   £££
££
(116, 4) (117, 4) (117, 4) (1, 5) (2, 5)
```
Notice how X is 117 two times. I assume on Windows, you'll see a `(0, 5)` for the second one.

With this change the visual behavior will now be the same as on Windows.

We could try to distinguish between the two `(117, 4)` but that brings extra complexity to the caching. Invalidating the cache is simpler.